source/unix/system_info.c

```diff
@@ -0,0 +1,21 @@
+/*
+* Copyright 2010-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+*
+* Licensed under the Apache License, Version 2.0 (the "License").
+* You may not use this file except in compliance with the License.
+* A copy of the License is located at
+*
+*  http://aws.amazon.com/apache2.0
+*
+* or in the "license" file accompanying this file. This file is distributed
+* on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+* express or implied. See the License for the specific language governing
+* permissions and limitations under the License.
+*/
+
+#include <aws/common/system_info.h>
+#include <unistd.h>
+
+size_t aws_system_info_processor_count (void) {
+    return (size_t) sysconf(_SC_NPROCESSORS_ONLN);
```

Comment:
sysconf returns -1 on error, which will be cast to max unsigned, which will pass the unit test